crates/red_knot_server/src/session/index.rs

```diff
@@ -72,10 +73,7 @@ impl Index {
     pub(crate) fn key_from_url(&self, url: Url) -> DocumentKey {
         if self.notebook_cells.contains_key(&url) {
             DocumentKey::NotebookCell(url)
-        } else if Path::new(url.path())
-            .extension()
-            .map_or(false, |ext| ext.eq_ignore_ascii_case("ipynb"))
-        {
+        } else if PySourceType::from(url.path()).is_ipynb() {
```

Comment:
this compares case-sensitively where the existing code compares case-insensitively. I don't actually know if that's correct for Jupyter notebooks, but I do know that casing _does_ matter for other Python files. `import foo` succeeds if the `foo` module is found at `foo.py`, but fails if the `foo` module is found at `foo.PY`

Reply:
Hard to get data on this...

<img width="1775" alt="Screenshot 2024-10-08 at 1 24 01 PM" src="https://github.com/user-attachments/assets/a5bb8b27-edc6-4a40-b06a-6a463760d275">

Reply:
Seems overkill to make the breaking change for `ipynb`, I think the `.PY` import logic does make sense though?

Reply:
> this compares case-sensitively where the existing code compares case-insensitively. I don't actually know if that's correct for Jupyter notebooks, but I do know that casing _does_ matter for other Python files. `import foo` succeeds if the `foo` module is found at `foo.py`, but fails if the `foo` module is found at `foo.PY`

Is this also true on case insensitive file systems? I do think that import resolver is different from what we consider a python file

Reply:
My experience is that most applications don't consider file extensions to be case sensitive. Naming a file test.txt or test.TXT doesn't change the application in which my desktop environment opens the file. That's why I think we shouldn't treat file extensions as case sensitive but I can see how this is beyond the scope of this pr

It does make sense that the module resolver only tests for the existence of <module>.py and whether <module>.PY is considered to match depends on the file system's case sensitivity